crates/typst/src/math/attach.rs

```diff
@@ -150,6 +164,44 @@ impl LayoutMath for Packed<PrimesElem> {
     }
 }
 
+/// Stretches a base to accommodate wide top or bottom attachments.
+///
+/// ```example
+/// $ H stretch(=)^"define" U + p V $
+/// $ P => Q stretch(=>, width: #200%) R $
+/// $ V stretch(->, width: #(100% + 1.5em))^("surjection") ZZ $
+/// ```
+#[elem(LayoutMath)]
+pub struct StretchElem {
+    /// The base to stretch.
+    #[required]
+    pub body: Content,
+
+    /// The width of the base to stretch to, relative to the maximum width of
+    /// the base and the attachments (if any).
+    pub width: Smart<Rel<Length>>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct StretchInfo {
+    from: Abs,
+    to: Smart<Rel<Length>>,
+}
```

Comment:
If the change in `impl LayoutMath for Packed<AttachElem>` is made then we can get rid of this struct as only `to` needs to be passed around on the `GlyphFragment`. The `from` field will always just be the fragment's width then.